jquery.xblock.js

```diff
@@ -370,8 +372,25 @@
                 }
             }
 
-            return this.init(options, root);
+            var initDeferred = this.init(options, root);
+            initDeferred.always(function() {
+                $this.toggleSpinner(root, false);
+            });
+            return initDeferred;
         },
+
+        toggleSpinner: function(root, show) {
+            var spinner_class = "xblock-spinner";
+            if (show) {
+                var spinner = $("<div/>").addClass(spinner_class);
+                var spinner_styling_wrapper = $("<div/>").addClass("spinner-wrapper").appendTo(spinner);
+                spinner_styling_wrapper.append($("<i/>").addClass('fa fa-spin fa-spinner'));
```

Comment:
@bradenmacdonald there's no request for this so far, and it would be trivial to implement when the need arises. So, I would stick to YAGNI here and not implement it right now.